plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/checkpoint/CheckpointWriter.java

```diff
@@ -85,10 +87,13 @@ public CheckpointWriter(TypeManager typeManager, CheckpointSchemaManager checkpo
 
     public void write(ConnectorSession session, CheckpointEntries entries, Path targetPath)
     {
+        boolean writeStatsAsJson = entries.getProtocolEntry().getMinWriterVersion() <= 2 || Boolean.parseBoolean(entries.getMetadataEntry().getConfiguration().getOrDefault(DELTA_CHECKPOINT_WRITE_STATS_AS_JSON_PROPERTY, "true"));
+        boolean writeStatsAsStruct = entries.getProtocolEntry().getMinWriterVersion() <= 2 || Boolean.parseBoolean(entries.getMetadataEntry().getConfiguration().getOrDefault(DELTA_CHECKPOINT_WRITE_STATS_AS_STRUCT_PROPERTY, "false"));
```

Comment:
I don't get this, if we use old writer we should write stats asJson/asStruct? I thought we need new writer for that ?
Also why do we have true for json and false for stats?

Reply:
> I don't get this, if we use old writer we should write stats asJson/asStruct? I thought we need new writer for that ?

I thought the below requirement & relevant explanation mean those properties should be ignored in old writers, but it seems Databricks respects the values regardless `minWriterVersion`. Hence, I removed the 1st condition.  

<br> | Requirements
-|-
Writer Version 3 | Enforce:<br>- `delta.checkpoint.writeStatsAsJson`<br>- `delta.checkpoint.writeStatsAsStruct`<br>- `CHECK` constraints

> Also why do we have true for json and false for stats?

The default values come from https://github.com/delta-io/delta/blob/master/PROTOCOL.md#checkpoint-format. I can add code comment if needed. 
```
* stats: Column level statistics can be stored as a JSON string in the checkpoint. This field needs to be written when statistics are available and the table property: delta.checkpoint.writeStatsAsJson is set to true (which is the default). When this property is set to false, this field should be omitted from the checkpoint.
* stats_parsed: The stats can be stored in their [original format](https://github.com/delta-io/delta/blob/master/PROTOCOL.md#Per-file-Statistics). This field needs to be written when statistics are available and the table property: delta.checkpoint.writeStatsAsStruct is set to true. When this property is set to false (which is the default), this field should be omitted from the checkpoint.
```

Reply:
Changed the both to true. Databricks seems writing `stats_parsed` field by default.

Reply:
Shouldn't we follow the protocol?